packages/pn-personagiuridica-webapp/src/pages/__test__/ToSAcceptance.page.test.tsx

```diff
@@ -68,4 +69,19 @@ describe('test Terms of Service page', () => {
     expect(result.container).toHaveTextContent(/tos.switch-label/i);
     expect(result.container).toHaveTextContent(/tos.button/i);
   });
+
+  it('tests the switch and button', () => {
+    const result = render(<ToSAcceptance tosConsent={tosFirstAcceptance} privacyConsent={privacyFirstAcceptance} />);
+
+    const switchElement = result.getByRole('checkbox');
+    const acceptButton = result.getByRole('button');
+
+    expect(acceptButton).toBeDisabled();
+
+    userEvent.click(switchElement);
+    expect(acceptButton).toBeEnabled();
+
+    userEvent.click(acceptButton);
+    expect(mockDispatchFn).toBeCalledTimes(1);
```

Comment:
For new tests, remove the mock of the dispatch and use the mockApi. You can see some examples in NewDeleghe.page.tsx